nixos/modules/services/networking/unbound.nix

```diff
@@ -84,31 +161,82 @@ in
         description = "Use and update root trust anchor for DNSSEC validation.";
       };
 
-      extraConfig = mkOption {
-        default = "";
-        type = types.lines;
-        description = ''
-          Extra unbound config. See
-          <citerefentry><refentrytitle>unbound.conf</refentrytitle><manvolnum>8
-          </manvolnum></citerefentry>.
+      settings = mkOption {
+        type = with types; submodule settingsSubmodule;
```

Comment:
To allow arbitrary clauses you can use
```nix
{
  type = attrsOf (coercedTo settingsType singleton (listOf settingsType));
}
```

This will require some change to the string generator though.

If you wanna get super fancy you can even use the recently introduced freeform modules (#82743) to add specific type checking and docs to a subset of supported clauses: https://hydra.nixos.org/build/125503151/download/2/nixos/index.html#sec-freeform-modules. Not needed though

Reply:
I think this has been resolved by using a freeform submodule.